torchx/schedulers/ray_scheduler.py

```diff
@@ -322,13 +323,25 @@ def wait_until_finish(self, app_id: str, timeout: int = 30) -> None:
                     break
                 time.sleep(1)
 
-        def _cancel_existing(self, app_id: str) -> None:  # pragma: no cover
+        def _parse_app_id(self, app_id: str) -> Tuple[str, str]:
+            # find index of '-' in the first :\d+-
+            m = re.search(r":\d+-", app_id)
+            if m:
+                sep = m.span()[1]
+                addr = app_id[: sep - 1]
+                app_id = app_id[sep:]
+                return addr, app_id
+
             addr, _, app_id = app_id.partition("-")
+            return addr, app_id
+
+        def _cancel_existing(self, app_id: str) -> None:  # pragma: no cover
+            addr, app_id = self._parse_app_id(app_id)
             client = JobSubmissionClient(f"http://{addr}")
             client.stop_job(app_id)
 
         def _get_job_status(self, app_id: str) -> JobStatus:
```

Comment:
It would be good to have explicit concepts that expresses app_id and  maybe_app_id_address.

Reply:
yeah, long term I'm not very happy with how we're adding the addr in here -- I think we might need to do some config refactors on this so there's a way to set a "scheduler" config for non submit methods. Most other schedulers pull context info like this from the environment variables but ray doesn't seem to do that